server/src/main/java/io/crate/types/ArrayType.java

```diff
@@ -173,7 +174,7 @@ public int compare(List<T> val1, List<T> val2) {
             return -1;
         }
         for (int i = 0; i < val1.size(); i++) {
-            int cmp = innerType.compare(val1.get(i), val2.get(i));
+            int cmp = Comparator.nullsFirst(innerType).compare(val1.get(i), val2.get(i));
```

Comment:
https://github.com/crate/crate/pull/9851
it should be fine probably to have a null-safe comparison here such as it the array type?

Reply:
yes should be fine. The `NULLS FIRST / LAST` would need  toapply for the array itself, not for individual elements in case we add support for sorting on arrays.